network-store-client/src/main/java/com/powsybl/network/store/client/RestNetworkStoreClient.java

```diff
@@ -56,7 +67,19 @@ public RestNetworkStoreClient(RestTemplateBuilder restTemplateBuilder) {
         }
     }
 
-    private  <T extends IdentifiableAttributes> List<Resource<T>> getAll(String target, String url, Object... uriVariables) {
+    private <T extends IdentifiableAttributes> void addAttributeSpyer(Resource<T> resource) {
+        resource.setResourceUpdater(resourceUpdater);
+
+        if (resource.getAttributes() instanceof AbstractAttributes) {
```

Comment:
In addition to this, we should enforce this at compile time in AttributeSpyer.java:

```java
    public static<T extends AbstractAttributes> T create(T attributes, ResourceType resourceType) {
```

Reply:
I think the generic parameter on the AttributeSpyer is not necessary anymore. Also we can add a private constructor and remove the supresswarning and rename AttributeSpyer to something like AttributeSpyerFactory (or AttributesInterceptorDirtyOnSettterFactory or something)

Reply:
> In addition to this, we should enforce this at compile time in AttributeSpyer.java:
> 
> ```java
>     public static<T extends AbstractAttributes> T create(T attributes, ResourceType resourceType) {
> ```

This is not possible as AbstractAttribute does not inherit from anything.

Reply:
> I think the generic parameter on the AttributeSpyer is not necessary anymore. Also we can add a private constructor and remove the supresswarning and rename AttributeSpyer to something like AttributeSpyerFactory (or AttributesInterceptorDirtyOnSettterFactory or something)

I removed the class generic parameter add a private constructor and remove rhe supresswarning.

I think  AttributeSpyer is correctly named. I just changed the method name to AttributeSpyer.spy().